mvc/views/view-components/sample/TodoList/src/TodoList/ViewComponents/PriorityListViewComponentAsync.cs

```diff
@@ -0,0 +1,46 @@
+﻿//#define Second
+#if Second
+
+using Microsoft.AspNet.Mvc;
+using System.Linq;
+using System.Threading.Tasks;
+using TodoList.Models;
+
+public class PriorityListViewComponent : ViewComponent
+{
+    private readonly ApplicationDbContext db;
+
+    public PriorityListViewComponent(ApplicationDbContext context)
+    {
+        db = context;
+    }
+
+    // Synchronous Invoke removed.
+
+    public async Task<IViewComponentResult> InvokeAsync(
+        int maxPriority, bool isDone)
+    {
+        var items = await GetItemsAsync(maxPriority, isDone);
+        return View(items);
+    }
+
+    private Task<IQueryable<TodoItem>> GetItemsAsync(
+        int maxPriority, bool isDone)
+    {
+        return Task.FromResult(GetItems(maxPriority, isDone));
```

Comment:
That's my question, but why are we showing something no one would do? I'd suggest the following:

```
private Task<IList<TodoItem>> GetItemsAsync(int maxPriority, bool isDone)
{
    return db.TodoItem.Where(x => x.IsDone == isDone && x.Priority <= maxPriority).ToListAsync();
}
```